### PR TITLE
DEVDOCS-4620: [new] Data Layer API, add new spec

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-4620]

This PR is related to [PR 1735](https://github.com/bigcommerce/dev-docs/pull/1735) (API doc) and [PR 174 ](https://github.com/bigcommerce-labs/next-dev-center-prototype/pull/174) (redirects)

[DEVDOCS-4620]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/data-layer-api.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
+# Data Layer API Reference
+
+The [Data Layer API](/api-docs/store-management/data-layer-api) is a GraphQL management API that allows you to enable and disable the [Big Open Data Layer](/api-docs/analytics/bodl-for-storefronts) for a store.
+
+**Request URL:** 
+
+`https://api.bigcommerce.com/stores/{store_hash}/v3/...`
```
Comment (bc-andreadao) [inline docs/data-layer-api.md:7]: @solofeed Do you have the request url?

File: docs/data-layer-api.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+| :--- | :--- | :--- | :--- |
+| `store_hash` | string | Unique ID for a store | Yes |
+
+## Authentication
+
+### OAuth scopes
+
+| UI Name | Permission | Parameter |
+| :--- | :--- | :--- |
+|  | modify |  |
```
Comment (bc-andreadao) [inline docs/data-layer-api.md:21]: @solofeed I assume the GraphQL DataLayer API authenticates using Store API Accounts (like the management APIs that are REST APIs)? Do you have the UI name and parameter for the OAuth Scopes?

File: docs/data-layer-api.md (hunk 1 above)
Comment (solofeed) [reply docs/data-layer-api.md:7]: This is the graphql endpoint
`stores/{store_hash}/graphql`

File: docs/data-layer-api.md (hunk 2 above)
Comment (solofeed) [reply docs/data-layer-api.md:21]: scope - Information
permission - Manage Web Analytics

**Review by solofeed — DISMISSED** [review]
LGTM!

File: docs/data-layer-api.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+| `store_hash` | string | Unique ID for a store | Yes |
+
+## Authentication
+
+### OAuth scopes
+
+| UI Name | Permission | Parameter |
+| :--- | :--- | :--- |
+| Information & Settings | modify | `store_v2_information` |
+| Information & Settings | read-only | `store_v2_information_read_only` |
```
Comment (solofeed) [inline docs/data-layer-api.md:22]: I have not checked how the read-only scope works for this endpoint
Have you tested this case?

Comment (bc-andreadao) [reply docs/data-layer-api.md]: I just checked it, and it turns out read-only doesn't work for the queries, so I decided to remove it. Thanks for pointing it out 👍

File: docs/data-layer-api.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>dataSolutions</strong></td>
+<td valign="top"><a href="#datasolutionssettings">DataSolutionsSettings</a></td>
+<td>
+
+Data solutions that is updated as a result of mutation.
```
Comment (bc-traciporter) [inline docs/data-layer-api.md:248]:
```suggestion
Data solutions that are updated as a result of mutation.
```

File: docs/data-layer-api.md [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>isDataLayerEnabled</strong></td>
+<td valign="top"><a href="#boolean">Boolean</a>!</td>
+<td>
+
+is data layer enabled
```
Comment (bc-traciporter) [inline docs/data-layer-api.md:273]:
```suggestion
Is data layer enabled?
```

File: docs/data-layer-api.md [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+
+### Scalars
+
+#### Boolean
+
+The `Boolean` scalar type represents `true` or `false`.
+
+#### String
+
+The `String` scalar type represents textual data, represented as UTF-8 character sequences. The String type is most often used by GraphQL to represent free-form human-readable text.
```
Comment (bc-traciporter) [inline docs/data-layer-api.md:288]:
```suggestion
The `String` scalar type represents textual data as UTF-8 character sequences. GraphQL most often uses the String type to represent free-form human-readable text.
```

File: docs/data-layer-api.md [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>isDataLayerEnabled</strong></td>
+<td valign="top"><a href="#boolean">Boolean</a>!</td>
+<td>
+
+Indicates if a data layer(bodl) object is enabled for all storefronts.
```
Comment (bc-traciporter) [inline docs/data-layer-api.md:135]:
```suggestion
Indicates enabled data layer (bodl) objects for all storefronts.
```

**Comment by bc-andreadao** [comment]
Closing this PR, as it is getting replaced by an autogenerated schema

**Pull request closed without merging**